server/src/main/java/io/unitycatalog/server/persist/CatalogRepository.java

```diff
@@ -159,6 +163,17 @@ && getCatalogDAO(session, updateCatalog.getNewName()) != null) {
         if (updateCatalog.getComment() != null) {
           catalogInfoDAO.setComment(updateCatalog.getComment());
         }
+        if (!updateCatalog.getProperties().isEmpty()) {
+          PropertyRepository.findProperties(session, catalogInfoDAO.getId(), Constants.CATALOG)
+              .forEach(session::remove);
+          session.flush();
+          PropertyDAO.from(updateCatalog.getProperties(), catalogInfoDAO.getId(), Constants.CATALOG)
+              .forEach(session::persist);
+
+          CatalogInfo catalogInfo = catalogInfoDAO.toCatalogInfo();
+          RepositoryUtils.attachProperties(
+              catalogInfo, catalogInfo.getId(), Constants.CATALOG, session);
```

Comment:
At line 180, we are returning catalogInfoDAO.toCatalogInfo(), so this change would have no effect, we should remove this and just do `RepositoryUtils.attachProperties(
              catalogInfoDAO.toCatalogInfo(), catalogInfo.getId(), Constants.CATALOG, session);` at line 180

Reply:
Since we want the properties irrespective of `if (!updateCatalog.getProperties().isEmpty())`